api/proto/config/appconfig.proto

```diff
@@ -26,6 +26,47 @@ enum MetaDataType {
   MetaDataDriveMultipart = 3; // Process multipart MIME for application
 }
 
+// The information about a snapshot
+message AvailableSnapshot {
+  string id = 1;
+}
+
+// The snapshot information for an Application Instance
+message SnapshotConfig {
+  // activeSnapshot is the id of the snapshot which is currently used.
+  // It may be empty if no snapshot is used. Worth to note that the field is
+  // not used alone to trigger a rollback. It just indicates that the device
+  // is currently using a snapshot with the given id. To trigger a rollback, in
+  // addition to it, the rollbackCounter should be increased. It is done to
+  // avoid unnecessary rollbacks when the same activeSnapshot is sent several
+  // times (we cannot relay on EVE to perform rollbacks after the first time
+  // it receives the corresponding message, so it's a regular case to send the
+  // same value several times). Also it is used to perform a rollback to the
+  // same snapshot several times.
+  string activeSnapshot = 1;
+  // rollbackCounter is necessary in addition to the activeSnapshot to request
+  // EVE to trigger a rollback. The rollbackCounter is incremented when a
+  // snapshot is used for a rollback. It should not be decreased.
+  uint32 rollbackCounter = 2;
```

Comment:
Just curious if we expect controller to persist this rollback counter ? Since it is supposed to be incremented only.

Reply:
Yeah, it should be handled by the controller. I think we have discussed it with @xyuria-zededa. @xyuria-zededa, fix me if I'm wrong.

Reply:
Ok just making sure.

Reply:
@OhmSpectator yes, i can confirm it